### ISSUE TITLE
Infinite loop on cy.getReact

### ISSUE THREAD

**Issue by Twiggeh** [abhinaba-ghosh/cypress-react-selector#136]
`cy.getReact('ModalButton').nthNode(0).should('exist').trigger('click');`

Produces this result, if getReact cannot find a component
![broken](https://user-images.githubusercontent.com/23149166/105880250-6fb17000-6003-11eb-9cc8-090fc6897e94.gif)

<details>

<summary>Whole it</summary>

```js

	it('Shows Modals for Buttons', () => {
		cy.visit('/');
		cy.waitForReact(1001);

		cy.getReact('MediumButtons')
			.should('exist')
			.each(node => {
				cy.wrap(node)
					.getProps('buttons')
					.then(buttonsProp => {
						for (
							let buttonIndex = 0;
							buttonIndex < buttonsProp.length;
							buttonIndex += 1
						) {
							const { btnName, modal, btnUrl }: Button = buttonsProp[buttonIndex];

							if (!modal) continue;

							cy.react('BtnListWrap', { props: { href: btnUrl } })
								.should('exist')
								.eq(0)
								.click()
								.then(() => {
									//cy.react('ModalWrapper').should('exist');
									cy.getReact('ModalButton').nthNode(0).should('exist').trigger('click');
								});
						}
					});
			});
	});
```
</details>

**Comment by Jsijbel** [comment]
I get the same result when using the following command.

`cy.getReact('*', { props: { value: { item: true } } })`

**Comment by maga-polito** [comment]
same result, on  cy.react('MyComponent').should('have.length', 1)

**Comment by abhinaba-ghosh** [comment]
Hi guys, thanks for reporting this behavior. I am looking into it. Can you share the `cypress.json`/ cypress configuration in this case?

**Comment by Twiggeh** [comment]
@abhinaba-ghosh [Here is the cypress.json](https://github.com/Twiggeh/Portfolio/blob/master/client/cypress.json)

**Comment by Jsijbel** [comment]
Our cypress.json:
```
{
  "env": {
    "TAGS": "not @ignore"
  },
  "ignoreTestFiles": ["*.js", "*.json", "**/disabled/**"],
  "testFiles": "**/*.{feature,features}",
  "reporter": "junit",
  "reporterOptions": {
    "mochaFile": "results/test-output-[hash].xml"
  },
  "viewportHeight": 900,
  "viewportWidth": 1280,
  "defaultCommandTimeout": 8000,

  "chromeWebSecurity": false,
  "baseUrl": "Our test url",
  "retries": {
    "runMode": 1,
    "openMode": 1
  }
}
```

**Comment by abhinaba-ghosh** [comment]
folks, I am unable to reproduce the bug with my tests and setup. However, I will pull down the repo from @Twiggeh to dig deeper. 

I am using cypress-react-selector `v.2.3.5`

**Comment by Jsijbel** [comment]
I noticed in your reproduction video you used the cy.react function and not the cy.getReact function. I had the problem with the getReact function. I believe cy.react() did error when the component was not found.


https://user-images.githubusercontent.com/39823861/106868848-93de1280-66cf-11eb-8ee6-513a48a838a0.mp4

**Comment by abhinaba-ghosh** [comment]
Thanks, I am able to re-produce the same.

**Comment by abhinaba-ghosh** [comment]
Hi, the fix is available in `v2.3.6`. Feel free to reopen the ticket if the issue still exists. 

On a different note, [Sponsor](https://www.buymeacoffee.com/abhinabaghosh) this project to prioritize the bug fixes and feature requests.
